synapse/api/presence.py

```diff
@@ -81,8 +81,18 @@ def as_dict(self) -> JsonDict:
         return attr.asdict(self)
 
     @staticmethod
-    def from_dict(d: JsonDict) -> "UserPresenceState":
-        return UserPresenceState(**d)
+    def from_db(
+        row: Tuple[str, str, int, int, int, Optional[str], bool]
+    ) -> "UserPresenceState":
+        return UserPresenceState(
+            user_id=row[0],
+            state=row[1],
+            last_active_ts=row[2],
+            last_federation_update_ts=row[3],
+            last_user_sync_ts=row[4],
+            status_msg=row[5],
+            currently_active=bool(row[6]),
+        )
```

Comment:
I'm concerned that this is so far away from where we declare the ordering of the columns in the SQL. Can we inline this?

Reply:
Inline it meaning -- return it directly from the datastore?

Reply:
There's only one use of this anyway, wonder why I bothered w/ a helper...